rosidl_typesupport_cpp/resource/action__type_support.cpp.em

```diff
@@ -63,3 +63,18 @@ get_action_type_support_handle<@('::'.join([package_name] + list(interface_path.
 }
 
 }  // namespace rosidl_typesupport_cpp
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+ROSIDL_TYPESUPPORT_CPP_PUBLIC
+const rosidl_action_type_support_t *
+ROSIDL_TYPESUPPORT_INTERFACE__ACTION_SYMBOL_NAME(rosidl_typesupport_cpp, @(', '.join([package_name] + list(interface_path.parents[0].parts) + [interface_path.stem])))() {
+  return ::rosidl_typesupport_cpp::get_action_type_support_handle<@('::'.join([package_name] + list(interface_path.parents[0].parts) + [interface_path.stem]))>();
+}
+
+#ifdef __cplusplus
+}
+#endif
```

Comment:
This whole thing should be within the `rosidl_typesupport_cpp` namespace declaration above.  That will also mean that you won't have to fully-qualify the call to `get_action_type_support_handle`.

Reply:
Sure, I did it this way to keep it consistent to how it is done in rosidl_typesupport_introspection_cpp:
https://github.com/ros2/rosidl/blob/master/rosidl_typesupport_introspection_cpp/resource/srv__type_support.cpp.em#L120

I personally prefer that way since the symbol will not be namespaced and thus it may avoid confusion but this is your call and if you could confirm that you want this inside the namespace, I will update this PR accordingly.

Reply:
Friendly push :)

Reply:
@clalancette @sloretz your confirmation is still needed so I can update the PR if necessary and we can merge this ASAP so I can release two ROS2 libraries.